src/callback.c

```diff
@@ -0,0 +1,140 @@
+#include "dieharder.h"	    
+#include <R_ext/Parse.h>
+
+#define CSTRING(string) CHAR(STRING_ELT(string, 0))
+typedef unsigned long int uli_t;
+
+static SEXP global_parsed_callback = NULL;
+
+static void callback_set_seed(void *unused, uli_t seed) {
+    // call R's set.seed using equivalent of parse(text="set.seed(seed)")
+    SEXP call = PROTECT(lang2(install("set.seed"), ScalarInteger(seed)));
```

Comment:
I tend to recommend against doing this.  Can you not call `set.seed()` from the R level?